### ISSUE TITLE
[Bug]: Unsupported engine error

### ISSUE THREAD

**Issue by yogarasu** [storybookjs/storybook#20793]
### Describe the bug

Since 7.0.0-beta.34 it shows Unsupported engine error.
This is due to engine version limitations of `node-abort-controller`.
`@storybook/builder-webpack5` depends on node-abort-controller.

```
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: node-abort-controller@3.1.0
npm ERR! notsup Not compatible with your version of node/npm: node-abort-controller@3.1.0
npm ERR! notsup Required: {"node":"<14.7.0"}
npm ERR! notsup Actual:   {"npm":"9.1.2"}

npm ERR! A complete log of this run can be found in:
```

### To Reproduce

https://github.com/revoltage-inc/tmp-nextjs

```shell
npm i

npm run update

npm run sb
```

### System

```shell
Environment Info:

  System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Max
  Binaries:
    Node: 18.12.1 - ~/.volta/tools/image/node/18.12.1/bin/node
    npm: 9.1.2 - ~/.volta/tools/image/npm/9.1.2/bin/npm
  Browsers:
    Safari: 16.2
  npmPackages:
    @storybook/addon-essentials: ^7.0.0-beta.34 => 7.0.0-beta.33 
    @storybook/addon-interactions: ^7.0.0-beta.34 => 7.0.0-beta.33 
    @storybook/addon-links: ^7.0.0-beta.34 => 7.0.0-beta.33 
    @storybook/addon-viewport: ^7.0.0-beta.34 => 7.0.0-beta.33 
    @storybook/nextjs: ^7.0.0-beta.34 => 7.0.0-beta.33 
    @storybook/react: ^7.0.0-beta.34 => 7.0.0-beta.33
```

**Comment by shilman** [comment]
Please upvote the underlying issue here https://github.com/southpolesteve/node-abort-controller/issues/39

**Comment by valentinpalkovic** [comment]
Please override the `node-abort-controller` version in your package.json as a workaround for now

```
 "overrides": {
    "node-abort-controller": "3.0.1"
  }
  ```

**Comment by shilman** [comment]
We are hoping that the underlying libraries (`node-abort-controller`, `fork-ts-webpack-plugin`) will issue a fix soon and are submitting a PR for that now.

In the meantime, there are a couple workarounds:
- overriding the package version per @valentinpalkovic 's suggestion above
- use yarn instead of npm (which ignores the engine field)

If we don't get a fix in the next day or so, we look at how to restructure Storybook to fix this.

**Comment by ndelangen** [comment]
Fix is likely happening today, upstream.

**Comment by yannbf** [comment]
Alright, we reached out to the maintainers and the issue has been fixed!
The transitive dependency `node-abort-controller` has released a fix in `3.1.1` so it should be good now (we tested on our end and the issue is gone). If you're experiencing this issue, please delete your package lock (and potentially node_module) files and rerun install. Thanks for your patience!
